core/src/main/java/inetsoft/web/admin/content/repository/ExportAssetService.java

```diff
@@ -0,0 +1,188 @@
+               }
+               catch(IOException e) {
+                  // Log or handle the exception during the close operation
+                  throw new RuntimeException("Failed to close BinaryTransfer stream", e);
+               }
+            }
+         }
+         catch(IOException e) {
+            throw new RuntimeException("Failed to create output stream while export asset file data", e);
+         }
+         return data;
+      }
+
+      return null;
+   }
+
+   @ClusterProxyMethod(FILE_LOCATION_CACHE_NAME)
+   public String getFileNameFromID(@ClusterProxyKey String exportID) {
+      return filePathMap.get(exportID);
+   }
+
+   private byte[] createExport(ExportedAssetsModel exportedAssetsModel, Principal principal)
+      throws Exception
+   {
+      String name = Tool.byteDecode(exportedAssetsModel.name());
+      boolean overwriting = exportedAssetsModel.overwriting();
+      List<SelectedAssetModel> entryData = exportedAssetsModel.selectedEntities();
+      List<RequiredAssetModel> assetData = exportedAssetsModel.dependentAssets();
+      List<XAsset> assets = deployService.getEntryAssets(entryData, principal);
+      List<PartialDeploymentJarInfo.SelectedAsset> entryDataArray = DeployUtil.getEntryData(assets);
+      assert assetData != null;
+      List<PartialDeploymentJarInfo.RequiredAsset> assetDataArray = assetData.stream()
+         .map(this::createRequiredAsset)
+         .collect(Collectors.toList());
+
+      PartialDeploymentJarInfo info = new PartialDeploymentJarInfo();
+      info.setName(name);
+      info.setDeploymentDate(new Timestamp(System.currentTimeMillis()));
+      info.setOverwriting(overwriting);
+      info.setSelectedEntries(entryDataArray);
+      info.setDependentAssets(assetDataArray);
+
+      byte[] zipFileBytes;
+
+      try(ByteArrayOutputStream out = new ByteArrayOutputStream()) {
+         DeployUtil.createExport(info, out);
+         zipFileBytes = out.toByteArray();
+      }
+      catch(Exception e) {
+         throw new Exception("Failed to get byte array of export jar file: " + e.getMessage());
+      }
+
+      return zipFileBytes;
+   }
+
+   private PartialDeploymentJarInfo.RequiredAsset createRequiredAsset(RequiredAssetModel model) {
+      PartialDeploymentJarInfo.RequiredAsset asset = new PartialDeploymentJarInfo.RequiredAsset();
+      asset.setPath(model.name());
+      asset.setType(model.type());
+      asset.setUser(model.user());
+      asset.setTypeDescription(model.typeDescription());
+      asset.setRequiredBy(model.requiredBy());
+      asset.setDetailDescription(model.detailDescription());
+      asset.setAssetDescription(model.assetDescription());
+      long lastModifiedTime = model.lastModifiedTime();
+
+      if(lastModifiedTime != 0) {
+         asset.setLastModifiedTime(lastModifiedTime);
+      }
+
+      return asset;
+   }
+
+   private final Map<String, byte[]> localFileContentCache;
```

Comment:
Why? This is going to blow up memory. Use a temp file and just store the temp file path.